docs/setup_gdb.md

```diff
@@ -8,520 +8,394 @@ GDB Tutorial
 ```
 (gdb) q
-A debugging session is active.
-
-  Inferior 1 [process 1090] will be killed.
-
-Quit anyway? (y or n) y
 ```
 
-## Input redirection
-Skip this subsection on your first time through the tutorial.  You can use input redirection to avoid typing program input each time you run (for debugging) a program.
+### Sanitizers
+We recommend enabling the address sanitizer and undefined behavior sanitizer. These will help you find memory errors like going off the end of an array or vector.
 
-Without input redirection, you can type input into GDB after running your program.
-```
-$ gdb main.exe
-...
-(gdb) r
-Starting program: /vagrant/src/eecs280/p1-stats-awdeorio/main.exe
-enter a filename
-main_test_data.tsv
-enter a column name
-B
-...
-```
+First, edit your `Makefile` and add the `CXXFLAGS` recommended by the [ASAN Quick Start](setup_asan.html#quick-start).
 
-With input redirection, use a run command like this `r < main_test.in`.
+### Input redirection
+If you're unfamiliar with input redirection, first read the CLI tutorial section on [input redirection](cli.html#input-redirection-).
+
+Run with input redirection.  Make sure to add the name of your input file (`main_test.in` in this example).
 ```
 $ gdb main.exe
 ...
 (gdb) r < main_test.in
+...
 ```
 
-## Arguments and options
-Skip this subsection for EECS 280 project 1.  You'll need it for project 2 and beyond.
-
-*Arguments* and *options* are inputs to a program typed at the command line.  Arguments are often required.  Options (AKA *flags* or *switches*) start with a hyphen (`-`), and are typically optional.
-
-**Arguments example** from project 2:  `resize.exe` is the name of the program, and the arguments are `horses.ppm`,  `horses_400x250.ppm`, `400`, and `250`.
+### Arguments and options
+Arguments and options are inputs to a program typed at the command line.  Here's an example from EECS 280 Project 5:
 ```console
-$ ./resize.exe horses.ppm horses_400x250.ppm 400 250
-$ gdb resize.exe
-(gdb) r horses.ppm horses_400x250.ppm 400 250
+$ ./main.exe train_small.csv test_small.csv --debug
 ```
+{: data-variant="no-line-numbers" data-highlight="1" }
+
+- `main.exe` is the name of the program
+- `train_small.csv` and `test_small.csv` are arguments
+- `--debug` is an option
 
-**Options example** from project 5:  `main.exe` is the name of the program.  `train_small.csv` and  `test_small.csv` are arguments.  `--debug` is an option.
+To run a program with options or arguments in LLDB, inc
 ```console
-$ ./main.exe train_small.csv test_small.csv --debug
 $ gdb main.exe
 (gdb) r train_small.csv test_small.csv --debug
 ```
+{: data-highlight="2" }
 
+## Debug
+In this section, we'll set a breakpoint, which pauses the debugger.  Then, we'll cover some of the options to continue execution.
 
-# Debug
-Start GDB with your executable.
-```console
-$ gdb stats_tests.exe
-GNU gdb (Ubuntu 9.2-0ubuntu1~20.04) 9.2
-...
-Reading symbols from stats_tests.exe...
-(gdb)
-```
+`n` **Next AKA Step Over**
```

Comment:
```suggestion
`n` **Next / Step Over:**
```
Unsure about AKA here and everywhere else. Thoughts on "Next / Step Over"? Same for all the other AKAs.
Also added colon for visual separateness.